src/basic/cap-list.c

```diff
@@ -64,3 +68,50 @@ int capability_from_name(const char *name) {
 int capability_list_length(void) {
         return (int) ELEMENTSOF(capability_names);
 }
+
+int capability_set_to_string_alloc(uint64_t set, char **s) {
+        _cleanup_free_ char *str = NULL;
+        unsigned long i;
+
+        assert(s);
+
+        for (i = 0; i < cap_last_cap(); i++)
+                if (set & (UINT64_C(1) << i)) {
+                        str = strjoina(str, capability_to_name(i));
+                        if (!str)
+                                return -ENOMEM;
+                }
+
+        *s = str;
+        str = NULL;
```

Comment:
you are returning a stack-allocated string here, this will reuslt in memory corruption